### ISSUE TITLE
WPF EnsureCoreWebView2Async never finishes on a computer without any exception or error

### ISSUE THREAD

**Issue by datvm** [MicrosoftEdge/WebView2Feedback#2506]
*Note that [this article](https://stackoverflow.com/questions/66829201/webview2-ensurecorewebview2async-never-returns) is not a duplicate. I am having a different issue.*

My client computer cannot see WebView2 content in any WPF project. After some debugging effort, I pinpointed it to `EnsureCoreWebView2Async` call. I sent him a brand new WPF project with these code:

```cs
    private async void Window_Loaded(object sender, RoutedEventArgs e)
    {
        this.wv.CoreWebView2InitializationCompleted += this.Wv_CoreWebView2InitializationCompleted;
        MessageBox.Show("EnsureCoreWebView2Async is about to run");
        await this.wv.EnsureCoreWebView2Async();
        MessageBox.Show("EnsureCoreWebView2Async finished");
    }

    private void Wv_CoreWebView2InitializationCompleted(object? sender, Microsoft.Web.WebView2.Core.CoreWebView2InitializationCompletedEventArgs e)
    {
        if (e.InitializationException is not null)
        {
            MessageBox.Show("EnsureCoreWebView2Async has exception: " + e.InitializationException);
        }
        else
        {
            MessageBox.Show("EnsureCoreWebView2Async successful: " + e.IsSuccess);
            this.wv.CoreWebView2.Navigate("https://www.example.com");
        }
    }
```

I even tried to wrap the code in `Window_Loaded` inside a `Dispatcher.Invoke()` call to make sure it's executed in a UI thread but it has the same result.

On all other computers that I tested (Windows 11 and 10), we see all messages that are supposed to show but on his computer (Windows 10, latest update) only "EnsureCoreWebView2Async is about to run" is showed. After that nothing happens at all. He has VS2022 installed and I tried debugging and stepping over `EnsureCoreWebView2Async` never ends.

We tried reinstalling .NET 6. Attempting to run WebView2 installation (Evergreen Bootstrapper download from [official website](https://developer.microsoft.com/en-us/microsoft-edge/webview2/)) results in message saying it's already installed.

I tried compiling on my computer and sent over to his computer, his computer would not show the website (call never finishes). I asked him to compile on his side and send over to me, my computer shows up the website just fine with all the messages showed so I am sure it's not because of the compiler/compiling environment.

So far I cannot find any second computer that has the same issue, but according to our Windows Store review, someone also has the issue (their UI is blank). I cannot find out what is causing the problem. Please advise us on how to debug this issue.

[AB#39941502](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/39941502)

**Comment by yunate** [comment]
@datvm Hi datvm, 
Could you help to check if the Edge brower can work well in his computer. If it works well, 
try to change the CoreWebView2CreationProperties::UserDataFolder to another folder, and try again. If it cannot work too, could you provide a screenshot of the task manager?

**Comment by datvm** [comment]
Hi, I tried moving to another folder (`E:\Temp\TestWv2`). Still have the same problem. Oh and this time we waited long enough that an exception is thrown due to timeout (`MessageBox` showed by `Wv_CoreWebView2InitializationCompleted`):

![image](https://user-images.githubusercontent.com/6388546/172896177-740b5fe0-7920-4693-8c30-395bfbd93664.png)

As for Task Manager screenshot, could you be more specific please? The process list is very long so I don't know what exactly to provide. Or is there anyway to dump a text list instead? Thanks!

**Comment by champnic** [comment]
@datvm It's possible the runtime just got into a broken state. If you want to you could try checking the installer logs and we could add that info to the bug:
https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics.md#installer--setup-logs

**Comment by yizhao5** [comment]
@datvm. Hello.
The printing of your sample code is correct. That is to say, the printing order is `"EnsureCoreWebView2Async is about to run"` -> `"EnsureCoreWebView2Async finished"`. And `"EnsureCoreWebView2Async has exception"` or `"EnsureCoreWebView2Async successful"` will not appear.

- Cause Analysis:

The function `EnsureCoreWebView2Async()` uses `_initTask` as a boolean switch to call `wv.CoreWebView2InitializationCompleted?.Invoke(*)`, and the latter one calls the function `Wv_CoreWebView2InitializationCompleted(*)`.

The response function will not be triggered each time in the sample code. Because when `EnsureCoreWebView2Async()` is called for the first time by constructor, the response function cannot be triggered because `wv.CoreWebView2InitializationCompleted` is not registered and is null. When `EnsureCoreWebView2Async()` is called for the second time by the user, although `wv.CoreWebView2InitializationCompleted` is registered at this time, the response function is skipped because the `_initTask` task is judged to have been completed.

So if you want to judge by waiting for the response function in your sample code, you can never wait.

- Example of fixing:

In short, you should register the response function in the constructor, so it will be triggered. As follows:
```cs
        public MainWindow()
        {
            InitializeComponent();
            this.wv.CoreWebView2InitializationCompleted += this.Wv_CoreWebView2InitializationCompleted;
        }
```

- Test Results: passed.

![image](https://user-images.githubusercontent.com/109595412/187895287-804a1992-2aaf-47aa-81ed-986f697bf504.png)

**Comment by yizhao5** [comment]
@datvm. Please check if it solves your problem. If so, please close it. Thanks for your cooperation.
